RELEASES_CHECKLIST.md

```diff
@@ -49,19 +51,36 @@ steps though, and we hope to make this more streamlined in the future.
 1. Open a release PR
     - Wait for approvals from Core team members.
     - Ensure the entire CI pipeline is green.
-1. Do a dry run with `cargo release [level] -v --no-tag --no-push`
-    - `[level]` will depend on what you're releasing.
+1. Do a dry run with `cargo release [new_version] -v --no-tag --no-push`
+    - `[new_version]` should be the **exact** SemVer compatible version you are attempting to release e.g. 
+      `4.0.0-alpha.3`
+      - It is possible to supply a SemVer level here e.g. `major`, `minor`, `patch` or `<pre-release-name>`, however
+        this will automatically bump and commit the changes to the `Cargo.toml` manifests. We have already done that in
+        an earlier step so it is not necessary.
     - We don't want `cargo-release` to create any releases or push any code, we'll do
        that manually once we've actually published to `crates.io`.
-1. If there are no errors, merge the release PR into `master`.
-1. Publish with `export PUBLISH_GRACE_SLEEP=5 && cargo release [level] -v --no-tag --no-push --execute`
+1. Check the output of the dry run:
+   - Does not show any automatic bumping of crate versions.
+   - Runs without error.
+1. Following a successful dry run, we can now publish to crates.io. This will be done from the release branch, since it
```

Comment:
Can you move the second and third sentences here down to another line? The first line should be an easy to digest "action item", and then we add more details after